packages/block-library/src/latest-posts/index.php

```diff
@@ -21,6 +21,12 @@ function render_block_core_latest_posts( $attributes ) {
 		'suppress_filters' => false,
 	);
 
+	$excerpt_length        = $attributes['excerptLength'];
+	$excerpt_length_filter = function ( $length ) use ( $excerpt_length ) {
+		return $excerpt_length;
+	};
+	add_filter( 'excerpt_length', $excerpt_length_filter, 999 );
```

Comment:
Why do we need a filter here?

Reply:
I used a filter to respect what the theme might do with the excerpt, and I don't trim later in the actual `get_the_excerpt` call so that manual excerpts are not trimmed.

Reply:
We are adding a filter but never removing the filter. Aren't we creating a bug where if the theme or plugin calls get_the_excerpt after the latest block is rendered we may be changing the length of that call?
It seems the render function of the latest blocks is having an unexpected side-effect of adding a filter and the filter may affect things outside of the latest post block that is being rendered.
Another concern I have is that I think in WordPress we should not add anonymous functions as filters, because it makes it impossible for a plugin to remove the filter. Maybe @azaozz can confirm if that's the case?

To me, it seems the safest bet, for now, is to call wp_trim_words on the result of get_the_excerpt without using a filter. We may trim even if the user wrote a manual excerpt but that seems expected because an excerptLength is being set also it was what happened before. So following this solution we still fix https://github.com/WordPress/gutenberg/issues/19418 without the risk of introducing a new bug in 5.4.
Any thoughts on this @draganescu, @youknowriad?

Reply:
Hey @jorgefilipecosta or I can remove the filter, after we get the contents and implement the proper behavior. Also good to know the detail on anonymous functions I didn't know that.

Reply:
> Hey @jorgefilipecosta or I can remove the filter, after we get the contents and implement the proper behavior


Hi @draganescu, does it mean not using a filter at all, or does it mean calling removing filter after it is not needed?
Either I think we need to update this PR soon so it can be part of Beta 3.

Reply:
I've updated the PR to not use an anonymous function but still use a filter. Also now the filter is removed after the block is rendered.